src/components/figures/chart/chartJs/zoomable_chart/zoomable_chartjs.ts

```diff
@@ -0,0 +1,476 @@
+import { useRef } from "@odoo/owl";
+import { Chart, ChartConfiguration, Scale } from "chart.js/auto";
+import {
+  MOVING_AVERAGE_TREND_LINE_XAXIS_ID,
+  TREND_LINE_XAXIS_ID,
+} from "../../../../../helpers/figures/charts/chart_common";
+import { Zone } from "../../../../../types";
+import { isCtrlKey } from "../../../../helpers/dom_helpers";
+import { chartJsExtensionRegistry, getChartJSConstructor } from "../chart_js_extension";
+import { ChartJsComponent } from "../chartjs";
+import { currentlyShownArea, hoveredPosition } from "./zoomable_chartjs_plugins";
+
+chartJsExtensionRegistry
+  .add("currentlyShownArea", currentlyShownArea)
+  .add("hoveredPosition", hoveredPosition);
+
+export class ZoomableChartJsComponent extends ChartJsComponent {
```

Comment:
would it be possible to move as much as possible the business logic to a store ? It would be infinitely easier to write tests, and also force us to decouple as much as possible from chartjs specifics.